packages/strapi-plugin-users-permissions/services/Providers.js

```diff
@@ -47,7 +47,7 @@ exports.connect = (provider, query) => {
       try {
         const user = await strapi.query('user', 'users-permissions').findOne({email: profile.email});
 
-        if (!strapi.plugins['users-permissions'].config.advanced.allow_register) {
+        if (!user && !strapi.plugins['users-permissions'].config.advanced.allow_register) {
```

Comment:
We should also verify if the providers is not `local`. Because, if I try to register with the same address than another user and the allow_register is set to false, I will be able to register whereas the registration is disabled?